models/MARBL_column/model_mod.f90

```diff
@@ -0,0 +1,638 @@
+! Reads the simulation length from a netCDF file.
+
+function read_model_time(filename)
+
+character(len=*), intent(in) :: filename
+type(time_type) :: read_model_time
+
+integer :: ncid
+character(len=*), parameter :: routine = 'read_model_time'
+real(r8) :: days
+type(time_type) :: mom6_time
+integer :: mom_base_date_in_days, mom_days
+
+mom_base_date_in_days = 139157 ! 1982 1 1 0 0
+ncid = nc_open_file_readonly(filename, routine)
+
+call nc_get_variable(ncid, 'Time', days, routine)
+
+call nc_close_file(ncid, routine)
+    
+mom_days = int(days) + mom_base_date_in_days
+
+read_model_time = set_time(0,mom_days)
+
+end function read_model_time
+
+!------------------------------------------------------------------
+! Returns the number of items in the state vector as an integer. 
+
+function get_model_size()
+
+integer(i8) :: get_model_size
+
+if ( .not. module_initialized ) call static_init_model
+
+get_model_size = model_size
+
+end function get_model_size
+
+!------------------------------------------------------------------
+! Given a state handle, a location, and a state quantity,
+! interpolates the state variable fields to that location and returns
+! the values in expected_obs. The istatus variables should be returned as
+! 0 unless there is some problem in computing the interpolation in
+! which case a positive istatus should be returned.
+
+subroutine model_interpolate(state_handle, ens_size, location, qty, expected_obs, istatus)
+
+type(ensemble_type), intent(in) :: state_handle
+integer,             intent(in) :: ens_size
+type(location_type), intent(in) :: location
+integer,             intent(in) :: qty
+real(r8),           intent(out) :: expected_obs(ens_size) !< array of interpolated values
+integer,            intent(out) :: istatus(ens_size)
+
+integer     :: qty_id, thickness_id, layer_index
+integer(i8) :: qty_index, thickness_index, ens_index
+real(8)     :: requested_depth, layerdepth_bottom, layerdepth_center, &
+               layerdepth_top, depth_below, depth_above, val_below, val_above
+real(8)     :: layer_thicknesses(nz, ens_size)
+real(8)     :: state_slice(ens_size)
+real(8)     :: loc_temp(3)
+
+if ( .not. module_initialized ) call static_init_model
+
+qty_id = get_varid_from_kind(state_dom_id, qty)
+
+! extracting the requested depth value from `location`.
+loc_temp = get_location(location)
+requested_depth = loc_temp(3)
+
+! extracting the current layer thicknesses for each ensemble member.
+thickness_id = get_varid_from_kind(state_dom_id, QTY_LAYER_THICKNESS)
+
+do layer_index = 1, nz
+    ! layer thicknesses are always extracted from the grid cell at index (1, 1), since the four grid
+    ! cells are supposed to represent identical columns.
+    thickness_index = get_dart_vector_index(1, 1, layer_index, state_dom_id, thickness_id)
+    layer_thicknesses(layer_index, :) = get_state(thickness_index, state_handle)
+end do
+
+
+! performing the interpolation for each ensemble member individually.
```

Comment:
This is a question/confirmation for my own understanding. There is no horizontal interpolation, in MARBL_column.
All four gridpoints have identical values for a given level so you only do vertical interpolation, and you just pick point (1,1).  You could pick (2,2) and the interpolation would give the same answer.

Reply:
Yes, you're right. All four corners are identical columns so technically you can select any corner. This also means that no horizontal interpolation is needed.